packages/project-management-automation/lib/add-first-time-contributor-label.js

```diff
@@ -1,50 +1,131 @@
+/**
+ * External dependencies
+ */
+const got = /** @type {*} */ ( require( 'got' ) ); // See: https://github.com/sindresorhus/got/issues/1137
+
 /**
  * Internal dependencies
  */
 const debug = require( './debug' );
+const getAssociatedPullRequest = require( './get-associated-pull-request' );
 
 /** @typedef {import('@actions/github').GitHub} GitHub */
-/** @typedef {import('@octokit/webhooks').WebhookPayloadPullRequest} WebhookPayloadPullRequest */
+/** @typedef {import('@octokit/webhooks').WebhookPayloadPush} WebhookPayloadPush */
+/** @typedef {import('./get-associated-pull-request').WebhookPayloadPushCommit} WebhookPayloadPushCommit */
+
+/**
+ * Base endpoint URL for WordPress.org profile lookup by GitHub username.
+ *
+ * @type {string}
+ */
+const BASE_PROFILE_LOOKUP_API_URL =
+	'https://profiles.wordpress.org/wp-json/wporg-github/v1/lookup/';
+
+/**
+ * Message of comment prompting contributor to link their GitHub account from
+ * their WordPress.org profile for props credit.
+ *
+ * @type {string}
+ */
+const ACCOUNT_LINK_PROMPT =
+	'Congratulations on your first merged pull request! We would like to ' +
+	'give you credit for your contribution in the next WordPress release, but ' +
+	'we were unable to find a WordPress.org profile associated with your ' +
+	'GitHub account. At your convenience, please visit the following URL and ' +
+	'click "link your GitHub account" under "GitHub Username" to initiate ' +
+	'the process to link your accounts:\n\nhttps://profiles.wordpress.org/me/profile/edit/\n\n' +
+	'If you do not have a WordPress.org account, you can create one at the ' +
+	'following page:\n\nhttps://login.wordpress.org/register';
 
 /**
- * Adds the 'First Time Contributor' label to PRs opened by contributors that
- * have not yet made a commit.
+ * Adds the 'First Time Contributor' label to PRs merged on behalf of
+ * contributors that have not yet made a commit.
  *
- * @param {WebhookPayloadPullRequest} payload Pull request event payload.
- * @param {GitHub}                    octokit Initialized Octokit REST client.
+ * @param {WebhookPayloadPush} payload Push event payload.
+ * @param {GitHub}             octokit Initialized Octokit REST client.
  */
 async function addFirstTimeContributorLabel( payload, octokit ) {
-	const owner = payload.repository.owner.login;
-	const repo = payload.repository.name;
-	const author = payload.pull_request.user.login;
+	if ( payload.ref !== 'refs/heads/master' ) {
+		debug(
+			'add-first-time-contributor-label: Commit is not to `master`. Aborting'
```

Comment:
Would it make sense to display the prompt regardless of branch? If they are opening a PR, it shows they are interested in contributing. Why not catch them the first time regardless of branch?

Reply:
>Would it make sense to display the prompt regardless of branch? If they are opening a PR, it shows they are interested in contributing. Why not catch them the first time regardless of branch?

Unfortunately, it's not possible, related to this point from the original comment:

>due to [permission constraints](https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#permissions-for-the-github_token) of GitHub Actions, labels cannot be applied by actions if the originating repository is a fork of Gutenberg.

A commit pushed to a branch originating from a fork would fall under these restrictions.

It works for commits to master, since it's within the source repository, committed by someone with permissions to push to master.